substrate/frame/election-provider-multi-block/src/signed/tests.rs

```diff
@@ -634,3 +637,49 @@ mod e2e {
 		})
 	}
 }
+
+mod defensive_tests {
+	use super::*;
+
+	#[test]
+	fn verification_data_unavailable_score_slashes_submitter() {
+		// Reporting VerificationDataUnavailable(Score) slashes the submitter
+		// Missing score means we reject the solution and slash the deposit
+		ExtBuilder::signed().build_and_execute(|| {
+			// Setup a leader first
+			roll_to_signed_open();
+			assert_eq!(balances(99), (100, 0));
+			assert_ok!(SignedPallet::register(RuntimeOrigin::signed(99), Default::default()));
+			assert_eq!(balances(99), (95, 5)); // deposit taken
+
+			// Simulate score unavailable - should slash the submitter
+			<SignedPallet as SolutionDataProvider>::report_result(
```

Comment:
Not a realistic scenario -- I suggest instead: 

* Properly call `Verifier::start`
* Mutate storage and delete the score of the leader (which is pretty impossible)